src/util/cmdlineargs.cpp

```diff
@@ -24,18 +24,24 @@ bool calcUseColorsAuto() {
     // see https://no-color.org/
     if (QProcessEnvironment::systemEnvironment().contains(QLatin1String("NO_COLOR"))) {
         return false;
-    } else {
+    }
+
 #ifndef __WINDOWS__
-        if (isatty(fileno(stderr))) {
-            return true;
-        }
+    if (!isatty(fileno(stderr))) {
+        return false;
+    }
 #else
-        if (_isatty(_fileno(stderr))) {
-            return true;
-        }
-#endif
+    if (!_isatty(_fileno(stderr))) {
+        return false;
     }
-    return false;
+#endif
+
+    // Check if terminal is known to support ANSI colors
+    QString term = QProcessEnvironment::systemEnvironment().value("TERM");
+    return term == "ansi" || term == "cygwin" || term == "linux" ||
+            term.startsWith("screen") || term.startsWith("xterm") ||
+            term.startsWith("vt100") || term.startsWith("rxvt") ||
+            term.endsWith("color");
```

Comment:
please also add `alacritty` and `xterm-256color`.

Reply:
`xterm-256color` is already handled by the last condition. Is `alacritty` the full string or is it suffixed with `-...color` too?

Reply:
It's the full string:

```
$ echo $TERM
alacritty
$
```

Reply:
on second thought, let's invoke `tput colors` to check for color support instead. If the command is not available, we just don't use colors.

Reply:
Yeah, that's probably a cleaner solution indeed. Only downside is that we'd have to spawn a process at every launch.

Reply:
I think we can live with that :)

Maybe something like this (untested):

```diff
diff --git a/src/util/cmdlineargs.cpp b/src/util/cmdlineargs.cpp
index f7b16684f0..356e9a0be3 100644
--- a/src/util/cmdlineargs.cpp
+++ b/src/util/cmdlineargs.cpp
@@ -9,6 +9,7 @@

 #include <QCommandLineParser>
 #include <QCoreApplication>
+#include <QProcess>
 #include <QProcessEnvironment>
 #include <QStandardPaths>
 #include <QtGlobal>
@@ -36,12 +37,32 @@ bool calcUseColorsAuto() {
     }
 #endif

-    // Check if terminal is known to support ANSI colors
-    QString term = QProcessEnvironment::systemEnvironment().value("TERM");
-    return term == "ansi" || term == "cygwin" || term == "linux" ||
-            term.startsWith("screen") || term.startsWith("xterm") ||
-            term.startsWith("vt100") || term.startsWith("rxvt") ||
-            term.endsWith("color");
+    // Check if terminal is support ANSI colors according to terminfo database
+    // using tput. If the command fails, don't use colors.
+    constexpr int kTimeoutMillis = 500;
+    QProcess tput;
+    tput.start("tput", {"colors"});
+    if (!tput.waitForStarted(kTimeoutMillis)) {
+        return false;
+    }
+
+    if (!tput.waitForFinished(kTimeoutMillis)) {
+        return false;
+    }
+
+    if (tput.exitStatus() != QProcess::NormalExit) {
+        return false;
+    }
+
+    if (tput.exitCode() != 0) {
+        return false;
+    }
+
+    if (tput.readAllStandardOutput().trimmed().compare("256") != 0) {
+        return false;
+    }
+
+    return true;
 }

 } // namespace
```

Reply:
We should check for platforms where we can't spawn processes (Wasm/iOS) too, but otherwise that looks good I'd say. I'll have a look at integrating it later

Reply:
Also I'm not sure if this would work on Windows (though the `TERM` heuristic probably wouldn't either), @JoergAtGithub do you by chance know if there's `tput` there or if there is another way of checking for colors?

Reply:
Sorry, I'm unfamilar with this topic

Reply:
On Windows I think we need to call [`GetConsoleMode()`](https://learn.microsoft.com/en-us/windows/console/getconsolemode) and check if the `ENABLE_VIRTUAL_TERMINAL_PROCESSING` flag is set.

Reply:
This could be useful:  https://github.com/build2/build2/issues/312#issuecomment-1693263055